pycode/memilio-simulation/memilio/simulation/math/integrator.h

```diff
@@ -0,0 +1,84 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Maximilian Betz
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef PYMIO_INTEGRATOR_H
+#define PYMIO_INTEGRATOR_H
+
+#include "memilio/math/integrator.h"
+#include "memilio/math/euler.h"
+#include "memilio/math/adapt_rk.h"
+#include "memilio/math/stepper_wrapper.h"
+#include <Eigen/Dense>
+#include "pybind_util.h"
+
+#include "pybind11/pybind11.h"
+#include <pybind11/eigen.h>
+
+namespace pymio
+{
+
+void bind_Integrator_Core(pybind11::module_& m)
+{
+    pybind11::class_<mio::IntegratorCore<double>, std::shared_ptr<mio::IntegratorCore<double>>>(m, "IntegratorCore")
```

Comment:
What is the second template argument `std::shared_ptr<mio::IntegratorCore<double>>` needed for?